src/NuGet.CommandLine/Commands/PushCommand.cs

```diff
@@ -53,6 +57,27 @@ public override void ExecuteCommand()
             }
         }
 
+        public async Task<string> GetPushEndpointAsync(string source)
+        {
+            var packageSource = new Configuration.PackageSource(source);
+
+            var sourceRepositoryProvider = new SourceRepositoryProvider(SourceProvider,
+                Enumerable.Concat(
+                    Protocol.Core.v2.FactoryExtensionsV2.GetCoreV2(Repository.Provider),
+                    Protocol.Core.v3.FactoryExtensionsV2.GetCoreV3(Repository.Provider)));
+
+            var sourceRepository = sourceRepositoryProvider.CreateRepository(packageSource);
+            var pushCommandResource = await sourceRepository.GetResourceAsync<PushCommandResource>();
```

Comment:
What's the experience for this like? Would you have to run:
`nuget.exe push -Source https://api.nuget.org/index.json {Package}`